src/sentry/api/serializers/models/group.py

```diff
@@ -883,7 +883,26 @@ def serialize(
         self, obj: Group, attrs: MutableMapping[str, Any], user: Any, **kwargs: Any
     ) -> BaseGroupSerializerResponse:
         result = super().serialize(obj, attrs, user)
-        del result["annotations"]  # type:ignore
+
+        ALLOWED_FIELDS = [
+            "culprit",
+            "id",
+            "isUnhandled",
+            "issueCategory",
+            "permalink",
+            "shortId",
+            "title",
+        ]
+        result = {k: v for (k, v) in result.items() if k in ALLOWED_FIELDS}
```

Comment:
I'm unfamiliar with what the frontend uses, but doesn't doing this delete many fields that are in the parent `GroupSerializer`? It looks like `BaseGroupSerializerResponse` expects many additional fields not in `ALLOWED_FIELDS`

Reply:
Right, we delete a bunch of unnecessary fields from the parent response. I checked all frontend usage, also tested manually -- seems working.
From typing point of view, I believe it should be fine because we have `total=False` ([totality](https://peps.python.org/pep-0589/#totality)):
https://github.com/getsentry/sentry/blob/0718ea6b4912d630af61edcde5747451833e9559/src/sentry/api/serializers/models/group.py#L141-L149

Reply:
Hmm I'm familiar with totality, but `BaseGroupSerializerResponse` expects other fields that are not being set as optional through `BaseGroupResponseOptional`.  I'm surprised there isn't a type error being thrown. I think the response type should be changed to `BaseGroupResponseOptional`  b/c the serializer can only be returning those optional fields

Reply:
Hmm I'm familiar with totality, but BaseGroupSerializerResponse expects other fields that are not being set as optional through BaseGroupResponseOptional like the original `assignedTo` which is being deleted through `ALLOWED_FIELDS`. I'm surprised there isn't a type error being thrown.

Reply:
The same situation was before when we were deleting `annotations` field. I'm not an expert in typing, but seems like the totality is inherited to a subclass, hence no fields are actually required in `BaseGroupSerializerResponse`?